### DISCUSSION TITLE
No permission to execute `AddProjectNextItem`

### DISCUSSION THREAD

**Discussion by michamos** [community/community#24601] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I’m trying to add many issues from a repository to a Github Project (beta). As there’s no way to do this in bulk through the website afaict, I tried using the API through the <code>gh</code> client as documented in <a href="https://docs.github.com/en/issues/trying-out-the-new-projects-experience/using-the-api-to-manage-projects" class="inline-onebox" rel="noopener nofollow ugc">Using the API to manage projects (beta) - GitHub Docs</a>. I managed to retrieve the list of issue IDs using</p>
<pre><code class="lang-auto">gh issue list -L 500 --json id -q ".[].id"
</code></pre>
<p>and the project ID using</p>
<pre><code class="lang-auto">gh api graphql -f query='
  query{    
    organization(login: "inspirehep"){                                                           
      projectNext(number: 59) {
        id
      }
    }
  }'
</code></pre>
<p>but when I’m trying to add an issue to that project following the documentation,</p>
<pre><code class="lang-auto">gh api graphql -f query='
  mutation {
    addProjectNextItem(input: {projectId: "PN_kwDOAAVxOM1mYw" contentId: "I_kwDOCaIres47pUhQ"}) {
      projectNextItem {
        id
      }
    }
  }'
</code></pre>
<p>I receive a permission error:</p>
<pre><code class="lang-auto">{
  "data": {
    "addProjectNextItem": null
  },
  "errors": [
    {
      "type": "FORBIDDEN",
      "path": [
        "addProjectNextItem"
      ],
gh: michamos does not have the correct permissions to execute `AddProjectNextItem`
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "message": "michamos does not have the correct permissions to execute `AddProjectNextItem`"
    }
  ]
}
</code></pre>
<p>I’m an admin of the organization and am able to add issues to the project through the web interface, but for some reason I don’t have permission to do it through the API.</p>
<p>Am I doing something wrong or is this a bug in the project (beta) API?</p>

**Comment by natikgadzhi** [comment]
<p>Running into the same problem ;( Did you figure this out yet?</p>

**Comment by michamos** [comment]
<p>No, it looks like a bug to me. I’ve just opened a <a href="https://support.github.com/ticket/personal/0/1407229" rel="noopener nofollow ugc">support ticket</a>.</p>

**Comment by bouk** [comment, marked as answer]
<p>I ran into this and figured it out: the default permissions that the GitHub CLI requests don’t give <code>write:org</code> permissions. You can fix this by generating your own PAT and pasting it in when running <code>gh auth login</code></p>

**Comment by michamos** [comment]
<p>Indeed, <a href="https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token" rel="noopener nofollow ugc">generating a PAT</a> with <code>write:org</code> permissions and passing that to the <code>gh auth login</code> did the trick. Thanks a lot <a class="mention" href="/bouk">@bouk</a>!</p>

**Comment by nickofthyme** [comment]
<p>Adding note for future people. This mutation is forbidden when using the GitHub <a href="https://docs.github.com/en/graphql/overview/explorer" rel="noopener nofollow ugc">GraphQL Explorer</a> with the built-in authentication.</p>

**Comment by nickofthyme** [comment]
<p>Same thing applies to <code>organization - projectV2</code> query…</p>
<pre data-code-wrap="graphql"><code class="lang-nohighlight">query {
    organization(login: "org"){
        projectV2(number: 123) {
            id
        }
    }
}
</code></pre>
<p>Error in the <a href="https://docs.github.com/en/graphql/overview/explorer" rel="noopener nofollow ugc">GraphQL Explorer</a> shows as…</p>
<pre><code class="lang-json">{
  "errors": [
    {
      "type": "INSUFFICIENT_SCOPES",
      "locations": [
        {
          "line": 3,
          "column": 9
        }
      ],
      "message": "Your token has not been granted the required scopes to execute this query. The 'projectV2' field requires one of the following scopes: ['read:project'], but your token has only been granted the: ['read:discussion', 'read:enterprise', 'read:gpg_key', 'read:org', 'read:packages', 'read:public_key', 'read:repo_hook', 'repo', 'user'] scopes. Please modify your token's scopes at: https://github.com/settings/tokens."
    },
    {
      "type": "INSUFFICIENT_SCOPES",
      "locations": [
        {
          "line": 4,
          "column": 13
        }
      ],
      "message": "Your token has not been granted the required scopes to execute this query. The 'id' field requires one of the following scopes: ['read:project'], but your token has only been granted the: ['read:discussion', 'read:enterprise', 'read:gpg_key', 'read:org', 'read:packages', 'read:public_key', 'read:repo_hook', 'repo', 'user'] scopes. Please modify your token's scopes at: https://github.com/settings/tokens."
    }
  ]
}
</code></pre>

**Comment by kustrun** [comment]
I didn't have correct permissions on Project to execute mutations. After switching to write mode it all worked as it should.
